src/subgraph.js

```diff
@@ -317,6 +318,76 @@ ${abiFunctions
       }, immutable.List())
   }
 
+  static validateBlockFunctions(manifest, { resolveFile }) {
+    return manifest
+      .get('dataSources')
+      .filter(
+        dataSource =>
+          dataSource.get('kind') === 'ethereum/contract' &&
+          dataSource.getIn(['mapping', 'blockHandlers'], immutable.List()).count() > 0,
+      )
+      .reduce((errors, dataSource, dataSourceIndex) => {
+        let path = ['dataSources', dataSourceIndex, 'blockHandlers']
+        // Use the Assemblyscript parser to generate an AST from the mapping file
+        let mappingFile = dataSource.getIn(['mapping', 'file'])
+        let mappingParser = new asc.Parser()
+        mappingParser.parseFile(
+          fs.readFileSync(resolveFile(mappingFile), 'utf-8'),
+          '',
+          false,
+        )
+
+        let blockHandlers = dataSource.getIn(
+          ['mapping', 'blockHandlers'],
+          immutable.List(),
+        )
+
+        // Ensure each blockHandler has a corresponding mapping handler
+        // with a compatible function signature
+        return errors.concat(
+          blockHandlers.reduce(
+            (errors, handler, index) =>
+              mappingParser.program.sources
+                .filter(source => source.kind == asc.SourceKind.DEFAULT)
+                .some(source =>
+                  source.statements
+                    .filter(
+                      statement => statement.kind === asc.NodeKind.FUNCTIONDECLARATION,
+                    )
+                    .some(
+                      functionDeclaration =>
+                        functionDeclaration.name.text === handler.get('handler') &&
+                        functionDeclaration.signature.parameters.length === 1 &&
+                        functionDeclaration.signature.parameters[0].name.text ==
+                          'block' &&
```

Comment:
Is it really productive to require a specific parameter name? Is it backward compatible?